pycloudlib/oci/utils.py

```diff
@@ -84,14 +82,126 @@ def get_subnet_id_by_name(
     return subnets[0].id
 
 
+def _get_subnet_features(
+    subnet: oci.core.models.Subnet,
+) -> Dict[str, Any]:
+    """
+    Get the availability domain, private, and networking type of the subnet.
+
+    Args:
+        subnet: The subnet model to get the features of.
+
+    Returns:
+        A dictionary containing the following keys:
+        - availability_domain: The availability domain of the subnet.
+        - private: Whether the subnet is private.
+        - networking_type: The networking type of the subnet
+
+    """
+    availability_domain = subnet.availability_domain
+    private = subnet.prohibit_internet_ingress
+    has_ipv4_cidr_block = subnet.cidr_block not in (None, "<null>")
```

Comment:
It might be worth putting the string `"<null>"` into a named variable at module scope to provide a bit of self-documentation for where this particular string comes from; at first glance it looks like an erroneously-applied javascript convention. I'm guessing this comes from the way the OCI api produces text/json output, but there's an opportunity here to make that more immediately clear.